Assets/MixedRealityToolkit.Providers/WindowsMixedReality/WindowsMixedRealityDeviceManager.cs

```diff
@@ -398,52 +402,79 @@ private void RegisterGestureEvents()
+            if (manipulationAction != MixedRealityInputAction.None)
+            {
+                gestureRecognizer.ManipulationStarted += GestureRecognizer_ManipulationStarted;
+                gestureRecognizer.ManipulationUpdated += GestureRecognizer_ManipulationUpdated;
+                gestureRecognizer.ManipulationCompleted += GestureRecognizer_ManipulationCompleted;
+                gestureRecognizer.ManipulationCanceled += GestureRecognizer_ManipulationCanceled;
+            }
 
-            gestureRecognizer.ManipulationStarted += GestureRecognizer_ManipulationStarted;
-            gestureRecognizer.ManipulationUpdated += GestureRecognizer_ManipulationUpdated;
-            gestureRecognizer.ManipulationCompleted += GestureRecognizer_ManipulationCompleted;
-            gestureRecognizer.ManipulationCanceled += GestureRecognizer_ManipulationCanceled;
+            if (selectAction != MixedRealityInputAction.None)
+            {
+                gestureRecognizer.Tapped += GestureRecognizer_Tapped;
+            }
         }
 
         private void UnregisterGestureEvents()
         {
             if (gestureRecognizer == null) { return; }
 
-            gestureRecognizer.HoldStarted -= GestureRecognizer_HoldStarted;
-            gestureRecognizer.HoldCompleted -= GestureRecognizer_HoldCompleted;
-            gestureRecognizer.HoldCanceled -= GestureRecognizer_HoldCanceled;
+            if (holdAction != MixedRealityInputAction.None)
+            {
+                gestureRecognizer.HoldStarted -= GestureRecognizer_HoldStarted;
+                gestureRecognizer.HoldCompleted -= GestureRecognizer_HoldCompleted;
+                gestureRecognizer.HoldCanceled -= GestureRecognizer_HoldCanceled;
+            }
 
-            gestureRecognizer.ManipulationStarted -= GestureRecognizer_ManipulationStarted;
-            gestureRecognizer.ManipulationUpdated -= GestureRecognizer_ManipulationUpdated;
-            gestureRecognizer.ManipulationCompleted -= GestureRecognizer_ManipulationCompleted;
-            gestureRecognizer.ManipulationCanceled -= GestureRecognizer_ManipulationCanceled;
+            if (manipulationAction != MixedRealityInputAction.None)
+            {
+                gestureRecognizer.ManipulationStarted -= GestureRecognizer_ManipulationStarted;
+                gestureRecognizer.ManipulationUpdated -= GestureRecognizer_ManipulationUpdated;
+                gestureRecognizer.ManipulationCompleted -= GestureRecognizer_ManipulationCompleted;
+                gestureRecognizer.ManipulationCanceled -= GestureRecognizer_ManipulationCanceled;
+            }
+
+            if (selectAction != MixedRealityInputAction.None)
+            {
+                gestureRecognizer.Tapped -= GestureRecognizer_Tapped;
+            }
         }
 
         private void RegisterNavigationEvents()
         {
-            if (navigationGestureRecognizer == null)
+            if (navigationAction != MixedRealityInputAction.None)
```

Comment:
I think that's reasonable. On a related note, it wasn't really clear to me why it's using separate gesture recognizers and switching between them vs having a single gesture recognizer and switching between manipulation/navigation settings using SetRecognizableGestures. IIRC, going into navigation mode also turns off hold (and now tap), which seems kind of weird - I understand why manipulate and navigate would be mutually exclusive, but it seems like hold/tap are useful in both cases.

Reply:
>IIRC, going into navigation mode also turns off hold (and now tap)

Oh huh, I thought we had done some tests a while back and found that wasn't the case, but I could be misremembering.

Reply:
Admittedly, I haven't tested it on device - it does work the way you describe in input sim, but input sim doesn't really work the same way as WMRDeviceManager.
But from looking at the code, WMRDeviceManager enforces that NavigationRecognizerEnabled and GestureRecognizerEnabled are mutually exclusive, and their corresponding gesture recognizers are started/canceled accordingly. In the default gesture profile, Manipulation Gestures includes tap and hold:
![image](https://user-images.githubusercontent.com/38891743/62257602-0b383500-b3bb-11e9-8f49-f6f3b1f40a4d.png)
but Navigation Gestures doesn't:
![image](https://user-images.githubusercontent.com/38891743/62257621-1b501480-b3bb-11e9-9e26-449ceedb9038.png)
That said, I don't seen anywhere that actually switches to navigation mode (and we don't use it for our app) so I don't know how important it is.

Reply:
The navigation gesture recognizer also doesn't register handlers for the tap/hold events, so I don't really see how it could be sending those events (unless the gesture recognizers just ignore CancelGestures and keep firing the events).

Reply:
I just tried enabling the navigation gesture recognizer (with default WMR gesture settings) using a WMR headset, and didn't get tap/hold events.

Reply:
also, I went ahead and added the check you requested initially (and fixed the merge conflict) - so hopefully this change is ready to go

Reply:
NIT: with regards to MixedRealityInputAction.None
You should also create a cached reference to this value instead of calling it each time, to save the cost of calling new.